### DISCUSSION TITLE
milvus_datanode_save_latency_bucket is too high

### DISCUSSION THREAD

**Discussion by navneetrajput06** [milvus-io/milvus#36652] [Q&A and General discussion]
Hi team,
During heavy insertion we are observing latency for below metrics. what could be the reason and what are possible cause of this latency.
We are suing S3 bucket as storage.
histogram_quantile(0.99, sum by (le, pod, node_id) (rate(milvus_datanode_save_latency_bucket{namespace="milvus"}[2m]))) >>> on each datanode avg value for this metrics is coming around 1000. 

please help me the understand the cause of this issue. Initially, when we starts it goes well but gradually its increasing.

**Comment by navneetrajput06** [comment]
@yhmo Kindly help.

**Comment by xiaofan-luan** [comment]
> Hi team, During heavy insertion we are observing latency for below metrics. what could be the reason and what are possible cause of this latency. We are suing S3 bucket as storage. histogram_quantile(0.99, sum by (le, pod, node_id) (rate(milvus_datanode_save_latency_bucket{namespace="isds-milvus"}[2m]))) >>> on each datanode avg value for this metrics is coming around 1000.
> 
> please help me the understand the cause of this issue. Initially, when we starts it goes well but gradually its increasing.

Check:
1. did you do manual flush while you insert data
2. if S3 has any quota limitation?
3. if you deploy graphana, what is the latency of each S3 Put?

Please send me an email if you need extra help james.luan@zilliz.com
We need detailed info like scale of your deployment, how you deploy you minio and so on.

**Reply by navneetrajput06** [reply]
did you do manual flush while you insert data. >>> No
if S3 has any quota limitation? >>> No
if you deploy graphana, what is the latency of each S3 Put? >>> What is the name of metrics to capture that ?

**Comment by navneetrajput06** [comment]
did you do manual flush while you insert data. >>> No
if S3 has any quota limitation?  >>> No
if you deploy graphana, what is the latency of each S3 Put?  >>> What is the name of metrics to capture that ?

**Reply by xiaofan-luan** [reply]
histogram_quantile(0.99, sum by (le, persistent_data_op_type) (rate(milvus_storage_request_latency_bucket{app_kubernetes_io_instance=~"$instance", app_kubernetes_io_name="$app_name", namespace="$namespace"}[2m])))

this is the metrics we are looking at. 
if you are using S3 then you should be able to see more metrics from AWS panel. 

I think for s3 if you don't do frequent flush it's hard to reach the quota limitation
